### ISSUE TITLE
Quoted procedure parameters cannot be parsed

### ISSUE THREAD

**Issue by zikato** [sqlfluff/sqlfluff#4969]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When a Snowflake procedure has a quoted parameter e.g.
```sql
CREATE OR REPLACE PROCEDURE myprocedure(
    "Id" NUMBER(38,0)
)
```

it will cause a parsing error. Removing the quotes around the parameter name fixes the problem.

Grabbing the DDL definition from Snowflake automatically wraps the parameters in quotes.

### Expected Behaviour

I'm expecting the code to parse

### Observed Behaviour

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE OR REPLACE
                       | PROCEDURE myprocedure(...'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?

### How to reproduce

CREATE OR REPLACE PROCEDURE myprocedure(
	"Id" NUMBER(38,0)
)
RETURNS VARCHAR
LANGUAGE SQL
AS
$$
-- Snowflake Scripting code
DECLARE
  radius_of_circle FLOAT;
  area_of_circle FLOAT;
BEGIN
  radius_of_circle := 3;
  area_of_circle := pi() * radius_of_circle * radius_of_circle;
  RETURN area_of_circle;
END;
$$
;

### Dialect

Snowflake

### Version

sqlfluff, version 2.1.2

### Configuration

[sqlfluff]

dialect = snowflake
templater = jinja
large_file_skip_byte_limit = 0

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
